core/arch/x86/mangle.c

```diff
@@ -1495,6 +1495,7 @@ mangle_direct_call(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
     if (target == (app_pc)retaddr) {
         LOG(THREAD, LOG_INTERP, 3, "found call to next instruction " PFX "\n", target);
     } else {
+        /* XXX i#3307: necessary to instr_set_translation_mangling_epilogue? */
```

Comment:
No, we can't skip any of the mangling after the suspend point: this should follow the other mangling and undo its push.

Reply:
Comment removed.